graviton/abi_strategy.py

```diff
@@ -183,3 +204,161 @@ def get(self) -> PyTypes:
+        """
+        Generates inputs appropriate for bare app calls and method calls
+        according to available argument_strategy.
+        """
+        assert (
+            self.argument_strategy
+        ), "cannot generate inputs without an argument_strategy"
+
+        mutating = self.abi_args_mod is not None
+
+        if not (method or mutating):
+            # bare calls receive no arguments (unless mutating)
+            return [tuple() for _ in range(self.num_dryruns)]
+
+        arg_types = self.argument_types(method)
+
+        prefix: List[bytes] = []
+        if self.handle_selector and method:
+            prefix = [self.method_selector(method)]
+
+        modify_selector = False
+        if (action := self.abi_args_mod) in (
+            ABIArgsMod.selector_byte_delete,
+            ABIArgsMod.selector_byte_insert,
+            ABIArgsMod.selector_byte_replace,
+        ):
+            assert (
+                prefix
+            ), f"{self.abi_args_mod=} which means we need to modify the selector, but we don't have one available to modify"
+            modify_selector = True
+
+        def selector_mod(prefix):
+            assert isinstance(prefix, list) and len(prefix) <= 1
+            if not (prefix and modify_selector):
+                return prefix
+
+            selector = prefix[0]
+            idx = random.randint(0, 4)
+            x, y = selector[:idx], selector[idx:]
+            if action == ABIArgsMod.selector_byte_insert:
+                selector = x + random.randbytes(1) + y
+            elif action == ABIArgsMod.selector_byte_delete:
+                selector = (x[:-1] + y) if x else y[:-1]
+            else:
+                assert (
+                    action == ABIArgsMod.selector_byte_replace
+                ), f"expected action={ABIArgsMod.selector_byte_replace} but got [{action}]"
+                idx = random.randint(0, 3)
+                selector = (
+                    selector[:idx]
+                    + bytes([(selector[idx] + 1) % 256])
+                    + selector[idx + 1 :]
+                )
+            return [selector]
+
+        def args_mod(args):
+            if action not in (ABIArgsMod.parameter_append, ABIArgsMod.parameter_delete):
+                return args
+
+            if action == ABIArgsMod.parameter_delete:
+                return args if not args else tuple(args[:-1])
+
+            assert action == ABIArgsMod.parameter_append
+            return args + (self.get(self.append_args_type),)
+
+        def gen_args():
+            # TODO: when incorporating hypothesis strategies, we'll need a more holistic
+            # approach that looks at relationships amongst various args
+            args = tuple(
+                selector_mod(prefix) + [self.get(atype) for atype in arg_types]
+            )
+            return args_mod(args)
+
+        return [gen_args() for _ in range(self.num_dryruns)]
+
+    def get(self, gen_type: abi.ABIType) -> PyTypes:
```

Comment:
nit: I didn't find usage of this method in this pr, so I would suggest taking it out. If it is (or will be) used in elsewhere, I suggest renaming to `get_abi_arg_by_strategy`, or something more specific.

Reply:
This is actually being used inside `generate()` in two list comprehensions in the local functions `args_mod()` and `gen_args()`.

In terms of naming. You bring up a good point. I think the current naming is confusing and vague. In particular, the _example generator_ methods:
* `generate` produces a `list[Sequence[PyTypes]]` which is supposed to be a list of _args-tuples_ with each _args-tuple_ intended as the input of a dry run
* `get` produces a single `PyTypes` 

One possible renaming would be:
* ~`generate()`~ $\rarr$ `generate_inputs()`
* ~`get()`~ $\rarr$ `generate_value()`

But I'm not totally convinced and am open to other suggestions.

Reply:
The possible renaming you proposed above makes sense to me.

Reply:
cf https://github.com/algorand/graviton/pull/49/commits/dfc2f7b0730006bc53f8dcd3013b92b7f8dd5052